newIDE/app/src/EventsFunctionsExtensionEditor/EventsFunctionConfigurationEditor/EventsFunctionParametersEditor.js

```diff
@@ -72,6 +74,80 @@ const validateParameterName = (i18n: I18nType, newName: string) => {
   return true;
 };
 
+type StrngSelectorEditorProps = {|
+  extraInfo: string,
+  setExtraInfo: newExtraInfo => void,
+|};
+
+const StringSelectorEditor = ({
+  extraInfo,
+  setExtraInfo,
+}: StrngSelectorEditorProps) => {
+  const [array, setArray] = React.useState([]);
+  const forceUpdate = useForceUpdate();
+  React.useEffect(
+    () => {
+      let parsedArray = null;
+      try {
+        parsedArray = JSON.parse(extraInfo);
+      } catch (e) {
+        console.warn('Cannot parse parameter extraInfo: ', e);
+      }
+
+      if (parsedArray !== null) {
+        setArray(parsedArray);
+      }
+    },
+    [extraInfo, array]
+  );
+
+  const updateExtraInfo = () => {
+    setExtraInfo(JSON.stringify(array));
+    forceUpdate();
+  };
+
+  return (
+    <ResponsiveLineStackLayout>
+      <Column justifyContent="flex-end" expand>
+        {array.map((item, index) => (
+          <Line key={index} justifyContent="flex-end" expand marginSize="5px">
+            <SemiControlledTextField
+              commitOnBlur
+              value={item}
+              onChange={text => {
+                array[index] = text;
+                updateExtraInfo();
+              }}
+              fullWidth
+            />
+            <IconButton
+              tooltip={t`Delete`}
+              onClick={() => {
+                array.splice(index, 1);
+                updateExtraInfo();
+              }}
+            >
+              <Delete />
+            </IconButton>
+          </Line>
+        ))}
+
+        <Line justifyContent="flex-end" expand>
+          <RaisedButton
+            primary
+            onClick={() => {
+              array.push('New String');
```

Comment:
nit: Let's call this `New Option`.

A string can be anything. An option is a word implying that you should be making a list of option for the user to choose.